conans/client/generators/cmake_common.py

```diff
@@ -281,15 +285,20 @@ def generate_targets_section(dependencies):
         endif()
     endif()
 
+    if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL ${CMAKE_SYSTEM_NAME})
+        set(CROSS_BUILDING 1)
+    endif()
+
+    # Avoid checks when cross compiling, apple-clang crashes because its APPLE but not apple-clang
     if( (CONAN_COMPILER STREQUAL "Visual Studio" AND NOT CMAKE_CXX_COMPILER_ID MATCHES MSVC) OR
         (CONAN_COMPILER STREQUAL "gcc" AND NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU") OR
-        (CONAN_COMPILER STREQUAL "apple-clang" AND (NOT APPLE OR NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")) OR
+        (CONAN_COMPILER STREQUAL "apple-clang" AND NOT CROSS_BUILDING AND (NOT APPLE OR NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")) OR
```

Comment:
Because is the only case that you are checking by OS and making assumptions about the compiler based on that.